internal/pkg/composable/providers/host/host.go

```diff
@@ -132,6 +132,7 @@ func ContextProviderBuilder(log *logger.Logger, c *config.Config, _ bool) (corec
 
 func getHostInfo(log *logger.Logger) func() (map[string]interface{}, error) {
 	return func() (map[string]interface{}, error) {
+		// We don't use the shared host info from util here, as we explicitly want the latest host information on every call.
```

Comment:
I would prefer if we didn't have two different ways to get Host information where callers need to think about which one to use, or whether they actually need the parts of it that can change at runtime (FQDN for an obvious example).

Can the periodic updates be pushed into `util.GetHost()`? They could be unconditional updates at a fixed rate, or it could behave as a rate limit where it updates at most every X seconds.

As a possible simplification for all of this, perhaps the network address information isn't valuable at all for containerized applications and we can add a way to filter it into https://github.com/elastic/go-sysinfo. There is an [IsContainerized()](https://github.com/elastic/go-sysinfo/blob/750e2c7a2db68de43d0b2ad7158f4e061de1a200/providers/linux/container.go#L30) already.

Reply:
For some related context, `add_kubernetes_metadata` stopped including the host IPs by default:
- https://github.com/elastic/integrations/issues/667

Reply:
Perhaps we could just filter out the link local addresses (assuming those are the problem here too) if just not having the addresses at all doesn't work, or its too risky to think being in a container always means you are in k8s (it doesn't). https://github.com/elastic/integrations/issues/6674#issuecomment-1609000789

Reply:
I feel like the periodicity of the host info checks in the provider should be a part of the provider's logic, as opposed to being hidden in the utils package. I could see adding a parameter to this API indicating if a cached value is fine, if that helps.

Honestly, what I'm really trying to do here is avoid recomputing the host info, which is only done at the start. All the other methods on this struct fetch the latest data anyway. Maybe the solution is to compute host info at call time, same as everything else? Then every component could have its own `types.Host` (as creating one would be cheap), and we'd separately cache host info for the packages that want it at startup.

Reply:
I think all I really care about is that it's obvious when a call to get host info will potentially contain stale data. Keeping the periodicity of the host info check in the provider is fine if that is clear.